static/app/utils/profiling/spanChart.tsx

```diff
@@ -0,0 +1,76 @@
+import {Rect} from 'sentry/utils/profiling/gl/utils';
+import {SpanTree} from 'sentry/utils/profiling/spanTree';
+import {makeFormatter} from 'sentry/utils/profiling/units/units';
+
+export interface SpanChartNode {
+  depth: number;
+  end: number;
+  node: SpanTree['spanTree'];
+  start: number;
+}
+
+class SpanChart {
+  spans: SpanChartNode[];
+  spanTree: SpanTree;
+
+  formatter = makeFormatter('milliseconds');
+  configSpace: Rect = Rect.Empty();
+  depth: number = 0;
+
+  constructor(spanTree: SpanTree, {configSpace}: {configSpace?: Rect} = {}) {
+    this.spanTree = spanTree;
+    this.spans = this.collectSpanNodes();
+
+    // @TODO once we start taking orphaned spans into account here we will need to
+    // modify this to min(start_timestamp) max(end_timestamp)
+    const duration =
+      spanTree.spanTree.span.timestamp - spanTree.spanTree.span.start_timestamp;
```

Comment:
`spanTree.spanTree.span` is referring to the placeholder node that is created as the root right? Where is its `start_timestamp` set?

Reply:
This was added in a next PR, I made the change to init the root span to the transaction so that it's no longer some arbitrary virtual root span